src/uwtools/config/tools.py

```diff
@@ -136,7 +126,7 @@ def _ensure_format(
     :raises: UWError if the format cannot be determined.
     """
     if isinstance(config, Config):
-        return FORMAT.yaml
+        return config.get_format()
```

Comment:
For the old supplemental files, we enforced that if `Config` objects were provided via the API, that they were `YAMLConfig` objects. We can loosen that requirement now and query the format from the `Config` object. Depth compatibility requirements are still enforced.